accounts/abi/bind/precompiletemplate.go

```diff
@@ -0,0 +1,312 @@
+{{if len .Normalized.Outputs | lt 1}}
+// Pack{{capitalise .Normalized.Name}}Output attempts to pack given [outputStruct] of type {{capitalise .Normalized.Name}}Output
+// to conform the ABI outputs.
+func Pack{{capitalise .Normalized.Name}}Output (outputStruct {{capitalise .Normalized.Name}}Output) ([]byte, error) {
+	return {{$contract.Type}}ABI.PackOutput("{{.Original.Name}}",
+		{{- range .Normalized.Outputs}}
+		outputStruct.{{capitalise .Name}},
+		{{- end}}
+  )
+}
+
+{{else if len .Normalized.Outputs | eq 1 }}
+{{$method := .}}
+{{$input := index $method.Normalized.Outputs 0}}
+// Pack{{capitalise .Normalized.Name}}Output attempts to pack given {{decapitalise $input.Name}} of type {{bindtype $input.Type $structs}}
+// to conform the ABI outputs.
+func Pack{{$method.Normalized.Name}}Output ({{decapitalise $input.Name}} {{bindtype $input.Type $structs}}) ([]byte, error) {
+	return {{$contract.Type}}ABI.PackOutput("{{$method.Original.Name}}", {{decapitalise $input.Name}})
+}
+{{end}}
+
+func {{decapitalise .Normalized.Name}}(accessibleState PrecompileAccessibleState, caller common.Address, addr common.Address, input []byte, suppliedGas uint64, readOnly bool) (ret []byte, remainingGas uint64, err error) {
+	if remainingGas, err = deductGas(suppliedGas, {{.Normalized.Name}}GasCost); err != nil {
+		return nil, 0, err
+	}
+
+  {{- if not .Original.IsConstant}}
+	if readOnly {
+		return nil, remainingGas, vmerrs.ErrWriteProtection
+	}
+ 	{{- end}}
+
+	{{- if len .Normalized.Inputs | eq 0}}
+	// no input provided for this function
+	{{else}}
+  // attempts to unpack [input] into the arguments to the {{.Normalized.Name}}Input.
+  // Assumes that [input] does not include selector
+	// You can use unpacked [inputStruct] variable in your code
+	inputStruct, err := Unpack{{capitalise .Normalized.Name}}Input(input)
+	if err != nil{
+		return nil, remainingGas, err
+	}
+	{{- end}}
+
+	{{if not .Original.IsConstant| and $contract.AllowList}}
+	stateDB := accessibleState.GetStateDB()
+	// Verify that the caller is in the allow list and therefore has the right to modify it
+	callerStatus := getAllowListStatus(stateDB, {{$contract.Type}}Address, caller)
+	if !callerStatus.IsEnabled() {
+		return nil, remainingGas, fmt.Errorf("%w: %s", ErrCannot{{.Normalized.Name}}, caller)
+	}
+  {{end}}
+	// CUSTOM CODE STARTS HERE
+	// use inputStruct ...
+	{{- if len .Normalized.Outputs | eq 0}}
+	// this function does not return an output, leave this one as is
+	packedOutput := []byte{}
+	{{- else}}
+	packedOutput, err := Pack{{.Normalized.Name}}Output(nil)
+	if err != nil {
+		return nil, remainingGas, err
+	}
+	{{- end}}
+
+	// Return an empty output and the remaining gas
+	return packedOutput, remainingGas, nil
+}
+{{end}}
+
+// create{{.Contract.Type}}Precompile returns a StatefulPrecompiledContract
+// with getters and setters for the precompile.
+{{if .Contract.AllowList}} //Access to the getters/setters is controlled by an allow list for [precompileAddr].{{end}}
```

Comment:
```suggestion
{{if .Contract.AllowList -}} // Access to the getters/setters is controlled by an allow list for [precompileAddr].{{end}}
```

Reply:
`-` is not necessarry here, I added the whitespace.